### PR TITLE
add --previousRun option to publish previous results without running

### PR DESCRIPTION
**Pull request by srosenberg**
This option was useful to publish partial results when a benchmark (in some package) suddenly crashes. E.g., 

```
benchdiff --new=99972ef --old=1831ab7 --count=10  --post-checkout='make buildshort' --sheets -p "2022-03-25T04_22_39Z"
Found previous run; old=benchdiff/1831ab7/artifacts/out.2022-03-25T04_22_39Z new=benchdiff/99972ef/artifacts/out.2022-03-25T04_22_39Z

generated sheet:  https://docs.google.com/spreadsheets/d/1ETobeUjrrpd91b_D_4qNAXWHGmUDN424Kjghbnk6DB0/edit
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -186,17 +190,39 @@ func run(ctx context.Context) error {
...
+		}
 
+		// Run the benchmarks.
+		tests := oldSuite.intersectTests(&newSuite)
+		err = runCmpBenches(ctx, &oldSuite, &newSuite, tests.sorted(), runPattern, itersPerTest)
+		if err != nil {
+			return err
+		}
+	} else {
+		// find output files for the given run
```
Comment (nvb) [inline main.go]: nit: The Go style guide requires proper capitalization and punctuation in comments. So:
```go
// Find output files for the given run.
```

File: main.go [hunk 2]
```diff
@@ -186,17 +190,39 @@ func run(ctx context.Context) error {
...
+		t, err := time.Parse(timeFormat, previousRun)
+
```
Comment (nvb) [inline main.go]: nit: the line breaks here are strange. They're usually after the error checking, not before. I would expect this to look like:
```go
		t, err := time.Parse(timeFormat, previousRun)
		if err != nil {
			return err
		}

		oldSuite.artDir = testArtifactsDir(oldSuite.ref)
		oldSuite.outFile, err = os.Open(oldSuite.getOutputFile(t))
		if err != nil {
			return err
		}

		newSuite.artDir = testArtifactsDir(newSuite.ref)
		newSuite.outFile, err = os.Open(newSuite.getOutputFile(t))
		if err != nil {
			return err
		}
```

**Review by nvb — APPROVED** [review]
LGTM, thanks!

File: main.go (hunk 1 above)
Comment (srosenberg) [reply main.go]: Fixing; it's taking longer than expected to undo the lower-casing habit :)

File: main.go (hunk 2 above)
Comment (srosenberg) [reply main.go]: Fixing; another habit from other PLs where it's preferred to separate control-flow from straight-line code.

**Comment by srosenberg** [comment]
TFTR!

**Pull request merged**